contrib/design-docs/config-file-parsing.md

```diff
@@ -0,0 +1,342 @@
+Merging the pull request will constitute approval by project maintainers to proceed with implementation work.
+When the feature is completed and merged, this document should be removed to avoid cluttering the repository.
+It will remain in the Git history for future retrieval if necessary.
+-->
+
+## **Short Summary**
+
+Unify (and rework) our config file parsing logic to make the various config files all behave
+the same parsing wise so users can better understand how it works.
+
+## **Objective**
+
+We have several config files such a containers.conf, storage.conf, registries.conf and more that
+get all implement their own parsing logic and have a different feature set. The goal is to
+consolidate the parsing into a separate package and then port all files to use that package
+instead making them behave consistently.
+
+## **Detailed Description:**
+
+### General
+
+Add new package to the storage library (`go.podman.io/storage/pkg/configfile`) which implements
+the core logic of how to read config files. The goal of the package is to define with config paths
+to use and in what order.
+
+It will however not define the structs and fields used for the actual content in each file, these
+stay where there are defined currently and the plan is to have the code there call into the `configfile`
+package to read the files in the same way.
+
+#### Search locations:
+
+Linux:
+
+- `/usr/share/containers/<name>.conf`
+- `/usr/share/containers/<name>.conf.d/`
+- `/usr/share/containers/<name>.rootful.conf.d/` (only when UID == 0)
+- `/usr/share/containers/<name>.rootless.conf.d/` (only when UID > 0)
+- `/usr/share/containers/<name>.rootless.conf.d/<UID>/` (only when UID > 0)
+
+- `/etc/containers/<name>.conf`
+- `/etc/containers/<name>.conf.d/`
+- `/etc/containers/<name>.rootful.conf.d/` (only when UID == 0)
+- `/etc/containers/<name>.rootless.conf.d/` (only when UID > 0)
+- `/etc/containers/<name>.rootless.conf.d/<UID>/` (only when UID > 0)
+
+- `$XDG_CONFIG_HOME/containers/<name>.conf`
+- `$XDG_CONFIG_HOME/containers/<name>.conf.d/`
+  (if $XDG_CONFIG_HOME is empty then it uses $HOME/.config)
+  This homedir lookup will also be done by root [1].
+
+Where `<name>` is `containers`, `storage` or `registries` for each config file.
+
+The `<name>.rootless.conf.d/<UID>/` is a directory named by the user id. Only the user with this
+exact uid match will read the config files in this directory.
+The use case is for admin to be able to set a default for a specific user without having to write
+into their home directory. Note this is not intended as security mechanism, the user home directory
+config files will still have higher priority.
+
+FreeBSD:
+
+Same as Linux except `/usr/share` is `/usr/local/share` and `/etc` is `/usr/local/etc`.
+
+Windows:
+
+There is no `/usr` equivalent, for `/etc` we instead lookup `ProgramData` env and use that one.
+And instead of `XDG_CONFIG_HOME` which isn't used on windows we use `APPDATA`.
+
+MacOS:
+
+Same as Linux.
```

Comment:
So, if I understand correctly, the Podman machine will have `XDG_CONFIG_HOME` set and will load configs from the host's filesystem through mounts. Am I correct?